server/src/main/java/org/opensearch/search/internal/ContextIndexSearcher.java

```diff
@@ -507,14 +507,18 @@ private boolean canMatch(LeafReaderContext ctx) throws IOException {
     }
 
     private boolean canMatchSearchAfter(LeafReaderContext ctx) throws IOException {
-        if (searchContext.request() != null && searchContext.request().source() != null) {
+        if (searchContext.searchAfter() != null
+            && searchContext.sort() != null
```

Comment:
- Any other component required besides sort? For example, aggregations, rescore?

Reply:
From what I can tell from the logic in `FieldSortBuilder.getMinMaxOrNullInternal`, this logic only applies to custom sorting, identifying opportunities for a `search_after` request to skip an entire segment if the lower/upper bound on the next hits lies outside the segment's bounds.

Reply:
`null` check for sort seems redundant since `getPrimaryFieldSortOrNull` will be null if the request doesn't have `sort`

Reply:
@noCharger Yeah, this is only applicable to sort. As sohami@ said, if we dont have sort in query, we wont invoke getMinMaxOrNull() function itself.
@sohami Makes sense. I cant think of any scenario where searchContext.sort() is null but request.source.sort is not.
removed the check